Added: TVT (15-31) Top Malo House

TheLocalPub opened the pull request:
My new mission.
Thought I had a PR for this last week, guess I forgot to do it. Only noticed now after checking in on it.
Very limited internet right now, will fix any issues (If there are any), as soon as I'm back to my pc.

Description: "A British Royal Marine unit has chosen to conduct a deliberate assault onto an Argentine OP, in order to destroy them, so not to delay follow on forces the next day."

joewhite94 reviewed (changes requested):
- Covermap has been reuploaded, use the one that already exists in the repo
- Time limit doesn't appear to be implemented
- Teleports remained unavailable after setup timer in local dedicated
- Although you're not using restriction zones, you could still set a freeze timer on the reforger lobby game mode to get the setup time displayed
- Be nice to copy over the GC_ARG faction to GC units after this is out, then you can remove the prefabs and config from here

TheLocalPub commented:
> * Covermap has been reuploaded, use the one that already exists in the repo
> * Time limit doesn't appear to be implemented
> * Teleports remained unavailable after setup timer in local dedicated
> * Although you're not using restriction zones, you could still set a freeze timer on the reforger lobby game mode to get the setup time displayed
> * Be nice to copy over the GC_ARG faction to GC units after this is out, then you can remove the prefabs and config from here

1. Added GC missions as a mod dependency to my pack so I can point the covermap to that image, thus allowing me to remove the covermap file I added.
2. Should now be implemented.
3. Teleports worked previously but I think where I added the interaction as a child to the flag pole itself before making this PR, it broke it I believe.
I've now removed the Tilw_interactions as children and just placed them directly into the layer, which works for me and allows me to teleport on each flag after 240 seconds.
4. Added. You are correct. While it won't affect gameplay in anyway, it would show a setup timer which is a good idea none the less.
5. I will do this once mission is live.

joewhite94 reviewed (changes requested):
other issues fixed, but teleport actions don't activate in dedicated server tool - I've asked Til to take a look when he has time

TheLocalPub commented:
> other issues fixed, but teleport actions don't activate in dedicated server tool - I've asked Til to take a look when he has time

Did he look?
@Til-Weimann

Til-Weimann commented:
Reproduced the issue.
The reason is that the availability check runs on the client, which doesn't know about the status of mission flags.
It cannot be moved to the server, so this cannot be fixed without replicating mission flags.
Unfortunately my time is pretty limited right now, but I might have another look in 2-3 weeks.

joewhite94 commented:
@TheLocalPub I recommend you put a restriction zone down that prevents UK from reaching the flags, and remove the condition from the teleport interaction - this is what I've done on the commandos remake and is similar to what people did in A3

Til-Weimann commented:
This should work, but make sure the teleport locations exist on the client. If they are streamable, it might not work.

TheLocalPub commented:
> This should work, but make sure the teleport locations exist on the client. If they are streamable, it might not work.

The location exist. I can TP to them myself when I test it local.
The locations chosen are generic entities.

TheLocalPub commented:
Are we sure teleporting works on a dedicated server?
Because as I said above, It works in local for me, the option to TP appears after the time expires, and It physically moves me to the generic entity.

joewhite94 commented:
> Are we sure teleporting works on a dedicated server? Because as I said above, It works in local for me, the option to TP appears after the time expires, and It physically moves me to the generic entity.

teleporting works on dedicated, the condition updating when a flag is set does not - the actions stay unavailable after setup timer

TheLocalPub commented:
> > Are we sure teleporting works on a dedicated server? Because as I said above, It works in local for me, the option to TP appears after the time expires, and It physically moves me to the generic entity.
> 
> teleporting works on dedicated, the condition updating when a flag is set does not - the actions stay unavailable after setup timer

Do we reckon this is something wrong with the teleport prefab? Or something I've done wrong in my missions event? - Although I'm confident I set it up correctly.

Til-Weimann commented:
It is nothing you did wrong, it's a fundamental issue with the script which cannot be fixed in just a few minutes.

TheLocalPub commented:
> It is nothing you did wrong, it's a fundamental issue with the script which cannot be fixed in just a few minutes.

Darn!

Well I'm glad I found yet another bug.
I guess this is the first time the teleport prefab was used?

Could you let me know once it's fixed please so I can update my mods and the missions :)

Til-Weimann commented:
The bug affects specifically checking the condition flag, without one it should still work correctly.

TheLocalPub commented:
> The bug affects specifically checking the condition flag, without one it should still work correctly.

Right, so in theory, I could remove the condition that enables the teleport, and it'll work, but then anyone could technically use the teleport before they are allowed?

I did set a restriction zone around the teleport flags, so if someone was to use the flag and teleported out the zone, would it teleport them back?
Another problem to this is that if they do teleport and the zone brings the user back, because they are on a platform in the sea, I fear it'd teleport them to the bottom of the ocean?

Til-Weimann commented:
TP issue will be fixed in next FW release

TheLocalPub commented:
> TP issue will be fixed in next FW release
Yay!

Will I need to update the missions PR?
